setup.py

```diff
@@ -496,6 +498,54 @@ def _read_requirements(filename: str) -> List[str]:
     return requirements
 
 
+def repackage_wheel(package_data: Dict[str, List[str]],
+                    wheel_location) -> None:
+    """Extracts libraries and other files from an existing wheel."""
+    assert _is_cuda(), "VLLM_USE_PRECOMPILED is only supported for CUDA builds"
+
+    import zipfile
+
+    if os.path.exists(wheel_filename := os.path.basename(wheel_location)):
+        print(f"Using existing wheel={wheel_filename}")
+    else:
+        try:
+            subprocess.check_call(
+                f"pip download --no-deps {wheel_location}".split(" "))
+        except subprocess.CalledProcessError:
+            # pip will not be available in PEP-517 style builds with
+            # build isolation, such as when running
+            #    `pip install <path|git+https://<url>`.
+            from urllib.request import urlretrieve
```

Comment:
when do we need `pip download`? I think we can always use `urllib`

Reply:
Because when it's available, it leverages the pip cache, which is useful to avoid re-downloading large wheels.

Reply:
in our case, we don't really want to cache the wheel file. the same url can point to different files. removed pip.